### PR TITLE
systemd/dracut decoupling

### PR DESCRIPTION
**Pull request by jpds**
Adds a clevis-luks-generic-unlocker for alternative use without systemd.

Based on patch by Sergio Correia

Closes: #346

### CODE REVIEW

File: src/dracut/clevis/clevis-luks-generic-unlocker [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -eu
+# vim: set ts=8 shiftwidth=4 softtabstop=4 expandtab smarttab colorcolumn=80:
+#
+# Copyright (c) 2020 Red Hat, Inc.
```
Comment (sarroutbi) [inline src/dracut/clevis/clevis-luks-generic-unlocker:5]: year might be changed to 2021

**Comment by Anonymous1157** [comment]
This patch needs a trivial change to also move the clevis-pin-null subfolder to src/dracut and then it will work for version 19.

Why isn't this PR merged after sitting for a year? What can I do to help get it merged? I am also using Clevis on OpenRC with no complaints other than maintaining this patch.

**Comment by gb-123-git** [comment]
@jpds @Anonymous1157 
Could you guys help me install clevis on Alpine? So far clevis works and decrypts any file, but I am not able to add it to initramfs/boot to unlock boot luks volume automatically using tang.  `dracut -f `command doesnt seem to detect clevis and automatic boot unlocking is not taking place.
Thanks guys ! 😄

**Comment by Anonymous1157** [comment]
@gb-123-git If you installed testing/clevis from Alpine's aports then you did not get the patch in this PR and it's not going to work no matter what you do. You have to write a new aport that applies this PR as a patch. (Or probably grab Gentoo's patch and use that because this one has yet to be rebased to latest and OP is probably not coming back to do it)

**Comment by sergio-correia** [comment]
> @Anonymous1157 I really dont know why this was not merged. How about I fork the github, apply the patch again after merging conflicts ? Will that work ? Will you help me test it ?
> 
> @npmccallum @sergio-correia Would you guys merge it this time if I really work on this? I dont want my many hours to go waste if this is not gonna get merged.

I am open to merging it, so if you want to work on it, please go ahead. Start by rebasing it on top of the current tree.

**Comment by gb-123-git** [comment]
@Anonymous1157 
Can you also test patch #437 ? You can build from ( https://github.com/gb-123-git/clevis/tree/master )

**Pull request closed without merging**